### ISSUE TITLE
Data List: Sending context to data source

### ISSUE THREAD

**Issue by abjerner** [leekelleher/umbraco-contentment#76]
Hi Lee,

Again, thanks for creating this package 👍 

Has it been considered to send a bit of information about the context of the data list to the data source? Eg. the ID of the page on which the data list has been inserted?

I know there are a whole lot of edge cases - eg. for detached elements, or just elements in general. But the context may be useful to filter or control the returned items depending on where in the site structure the data list has been inserted. Eg. a if the data list is used for picking categories, the data list could be limited to categories of the particular site only.

I had a quick look at the source code, and I fear this isn't possible without creating a breaking change. Possibly a special key in the configuration could be used for the context?

https://github.com/leekelleher/umbraco-contentment/blob/develop/src/Umbraco.Community.Contentment/DataEditors/DataList/IDataListSource.cs#L12

So to summarize:

### Is your feature request related to a problem? Please describe.

Context (current page) isn't passed on to the data source.

### Describe the solution you'd like

The context (primary ID, but maybe other information as well) would be passed on to the data source letting developers better control the returned items.

If you can point me in the right direction, I'd be happy to get my hands dirty, and work on a PR.

### Describe alternatives you've considered

If not possible in Contentment, I've considered creating my own data list package. It may not be a 100% clone, but something that fits our needs to @skybrud.

### Additional context

Keep up the good work 👍

**Comment by abjerner** [comment]
Oh - I tried to look for existing issues, but didn't find one. But just now my colleague send me a link to https://github.com/leekelleher/umbraco-contentment/issues/30

I think that answers this request then. But I'll let the issue stay open to hear your thoughts for this issue as well 😉

**Comment by abjerner** [comment]
@leekelleher thanks for the detailed answer 😄 

I think my colleague got something to work by reading from the query string. I'm not really sure yet about our approach from here. Maybe that is sufficient for now. Maybe we'll try to build something from Contentment. Or build something from scratch to fit our particular use case as we don't really need all the flexibility of the data list (we'll still be using Content Vanilla Edition for lots of other scenarios)

**Comment by leekelleher** [comment]
Thanks @abjerner. I'll close off this ticket (for now). 👍
